app/services/user_destroyer.rb

```diff
@@ -49,7 +51,11 @@ def destroy(user, opts = {})
             end
           end
 
-          PostDestroyer.new(@actor.staff? ? @actor : Discourse.system_user, post).destroy
+          if post.is_first_post? && category_topic_ids.include?(post.topic_id)
```

Comment:
```suggestion
          if post.is_first_post? && Category.exists?(topic_id: post.topic_id)
```

Reply:
Nice! I didn't know about `exists?` syntax.

Nonetheless, I think we should stay with the current version of first gathering ids and then checking. This code is in a loop so a user with many posts may cause a lot of additional queries. 

```ruby
        user.posts.each do |post|
```
@tgxworld what do you think?

Reply:
Ahhh Github diff didn't display the loop.....